### PR TITLE
Exposed contact_dialog methods to Lua

### PR DESCRIPTION
**Pull request by ZopharShinta**
Added calls from C++ into lua for entity_interact and on_dialog_button
Lua script for entities would be entities.lua in the same folder as locations.lua and plaques.lua for the current zone

I couldn't get the QHash for contactLinkHash in Contact.h to work, so i'm sending the int from Lua instead of the string. Example (3 vs CONTACTLINK_BYE)

### CODE REVIEW

**Comment by broxen** [comment]
In discord you mentioned adding more to this PR. I'll wait until I hear from you to pull down and test.

Thanks @ZopharShinta you rock!

**Comment by ZopharShinta** [comment]
Just for documentation forward. I added the following functions with the latest commit.

Character.giveXp(client, int xp
Character.giveDebt(client, int debt)
Character.giveHp(client, float hp)
Character.giveEnd(client, float end)
Character.giveInf(client, int inf)
Character.giveInsp(client, string inspirationName)
Character.giveEnhancement(client, string enhancementName)

File: Projects/CoX/Servers/MapServer/ScriptingEngine.cpp [hunk 1]
```diff
@@ -86,14 +91,178 @@ void ScriptingEngine::registerTypes()
...
             cl->addCommandToSendNextUpdate(std::unique_ptr<StandardDialogCmd>(n));
+        },
+        "browser", [](MapClientSession *cl, const char *content){
+            cl->addCommand<Browser>(content);
+        },
+        "contact_dialog",[](MapClientSession *cl, const char *message, sol::as_table_t<std::map<std::string, sol::as_table_t<std::vector<std::string>>>> buttons){
+            std::vector<ContactEntry> active_contacts;
+            const auto& listMap = buttons.source;
+
+            for (const auto& kvp : listMap) {
```
Comment (broxen) [inline Projects/CoX/Servers/MapServer/ScriptingEngine.cpp]: Can we please put brackets on their own line per our [coding styleguide](https://github.com/Segs/Segs/blob/develop/docs/CONTRIBUTING.md#coding-styleguide)

File: Projects/CoX/Servers/MapServer/ScriptingEngine.cpp [hunk 2]
```diff
@@ -86,14 +91,178 @@ void ScriptingEngine::registerTypes()
...
+            QString message = FloatingInfoMsg.find(fInfoMessage).value();
+            cl->addCommand<FloatingInfo>(cl->m_ent->m_idx, message, FloatingInfo_Attention , 4.0);
         }
+        );
+
+    m_private->m_lua.new_usertype<Character>("Character",
+
+
+    "giveDebt", [](MapClientSession *cl, int debt){
+        uint32_t currentDebt = getDebt(*cl->m_ent->m_char);
```
Comment (broxen) [inline Projects/CoX/Servers/MapServer/ScriptingEngine.cpp]: Can we please rename variables `in_our_underscore_format` following our [coding styleguide](https://github.com/Segs/Segs/blob/develop/docs/CONTRIBUTING.md#coding-styleguide)

Thank you!

File: Projects/CoX/Servers/MapServer/ScriptingEngine.cpp [hunk 3]
```diff
@@ -86,14 +91,178 @@ void ScriptingEngine::registerTypes()
...
+
+        uint32_t currentDebt = getDebt(*cl->m_ent->m_char);
+        if(currentDebt > 0){
+            uint32_t debtToPay = 0;
+            uint32_t halfXp = xp / 2;
+            if(currentDebt > halfXp){
+                debtToPay = halfXp; //Half to debt
+                xp = halfXp;
+
+            }else{
```
Comment (broxen) [inline Projects/CoX/Servers/MapServer/ScriptingEngine.cpp]: please put else statements on their own line per our [coding styleguide](https://github.com/Segs/Segs/blob/develop/docs/CONTRIBUTING.md#coding-styleguide)

Thank you!

**Comment by ZopharShinta** [comment]
Done. I corrected the format and added more to Lua.
Added set_target call into Lua
Added sendFloatingDamage function to Lua

**Pull request merged**